test/gtest/uct/ib/test_ib_event.cc

```diff
@@ -0,0 +1,105 @@
+/**
+* Copyright (C) Mellanox Technologies Ltd. 2020.  ALL RIGHTS RESERVED.
+* See file LICENSE for terms.
+*/
+
+#include <infiniband/verbs.h>
+
+extern "C" {
+#include <uct/ib/mlx5/ib_mlx5.h>
+#include <uct/ib/rc/accel/rc_mlx5.h>
+#include <uct/ib/rc/verbs/rc_verbs.h>
+}
+
+#include <uct/uct_p2p_test.h>
+
+class uct_p2p_test_event : public uct_p2p_test {
+public:
+    uct_p2p_test_event(): uct_p2p_test(0) {}
+
+    static ucs_log_level_t orig_log_level;
+    static volatile unsigned flushed_qp_num;
+
+    static ucs_log_func_rc_t
+    last_wqe_check_log(const char *file, unsigned line, const char *function,
+                       ucs_log_level_t level,
+                       const ucs_log_component_config_t *comp_conf,
+                       const char *message, va_list ap)
+    {
+        std::string msg = format_message(message, ap);
+        int ret;
+
+        ret = sscanf(msg.c_str(),
+                     "IB Async event on %*s SRQ-attached QP 0x%x was flushed",
+                     &flushed_qp_num);
+        if (ret) {
+            UCS_TEST_MESSAGE << msg.c_str();
+        }
+
+        return (level <= orig_log_level) ? UCS_LOG_FUNC_RC_CONTINUE
+            : UCS_LOG_FUNC_RC_STOP;
+    }
+
+    int wait_for_last_wqe_event(entity &e) {
+        const resource *r = dynamic_cast<const resource*>(GetParam());
+        flushed_qp_num = -1;
+        uint32_t qp_num;
+
+        if (r->tl_name == "rc_mlx5") {
+            uct_rc_mlx5_ep_t *ep = (uct_rc_mlx5_ep_t *)e.ep(0);
+            uct_ib_mlx5_qp_t *qp = &ep->tx.wq.super;
+            qp_num = qp->qp_num;
+            if (qp->type == UCT_IB_MLX5_OBJ_TYPE_DEVX) {
+                uct_ib_mlx5_devx_modify_qp_state(qp, IBV_QPS_ERR);
+            } else {
+                uct_ib_modify_qp(qp->verbs.qp, IBV_QPS_ERR);
+            }
```

Comment:
always call uct_ib_mlx5_modify_qp_state()

Reply:
uct_ib_mlx5_modify_qp_state() defined in ib_mlx5.inl
ib_mlx5.inl wan't compile in C++ environment so it can't be include here

Reply:
can we move it to C file?